packages/zilliqa-js-crypto/src/zilAddress.ts

```diff
@@ -0,0 +1,103 @@
+import { validation } from '@zilliqa-js/util';
+import {
+  isValidChecksumAddress,
+  decodeBase58,
+  toChecksumAddress,
+  encodeBase58,
+} from './util';
+import { toBech32Address, fromBech32Address } from './bech32';
+
+/**
+ * @enum AddressType
+ */
+enum AddressType {
+  bytes20 = 'bytes20',
+  bytes20Hex = 'bytes20Hex',
+  checkSum = 'checkSum',
+  base58 = 'base58',
+  bech32 = 'bech32',
+}
+
+class ZilAddress {
+  raw: string;
+
+  addressType?: AddressType;
+
+  bytes20Hex?: string;
+
+  bytes20: string;
+
+  checkSum?: string;
+
+  bech32?: string;
+
+  base58?: string;
+
+  constructor(raw: string) {
+    this.raw = raw;
+    this.bytes20 = '0x';
+    this.getAddressType();
+  }
+
+  isBytes20() {
+    return this.addressType === AddressType.bytes20;
+  }
+  isBytes20Hex() {
+    return this.addressType === AddressType.bytes20Hex;
+  }
+  isChecksum() {
+    return this.addressType === AddressType.checkSum;
+  }
+  isBase58() {
+    return this.addressType === AddressType.base58;
+  }
+  isBech32() {
+    return this.addressType === AddressType.bech32;
+  }
+
+  private getAddressType() {
+    const addrBool = validation.isAddress(this.raw);
+    const base58Bool = validation.isBase58(this.raw);
+    const bech32Bool = validation.isBech32(this.raw);
+    const checksumBool = isValidChecksumAddress(this.raw);
+
+    if (addrBool === true && checksumBool === false) {
+      this.addressType = AddressType.bytes20;
+      this.bytes20 = this.raw.startsWith('0x')
+        ? this.raw.substring(2)
+        : this.raw;
+      this.normalize();
+    } else if (addrBool === true && checksumBool === true) {
```

Comment:
Please early return instead of using multiple consecutive `if/else` blocks. E.g.

```
if (validation.isAddress(this.raw) { this.normalize(); return; }
if (validation.isBase58(this.raw) { this.normlize(); return; }
...
```

Reply:
Lesson learned! thanks!